### PR TITLE
feat(AM-11526,AM-11741): Slice Gateway External LoadBalancer 

### PR DESCRIPTION
**Pull request by mridulgain**
# Description

Controller changes for Slice Gateway External LoadBalancer implementation
- introduced `SliceGatewayServiceType` in `SliceConfig` & `WorkerSliceConfig`

Fixes #

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] Does this PR requires documentation updates?
* [X] I've updated documentation as required by this PR.
* [X] I have performed a self-review of my own code.
* [X] I have commented my code, particularly in hard-to-understand areas.
* [X] I have tested it for all user roles.
* [ ] I have added all the required unit test cases.

## Does this PR introduce a breaking change for other components like worker-operator?

```release-note

```

### CODE REVIEW

File: service/worker_slice_config_service.go [hunk 1]
```diff
@@ -211,6 +212,22 @@ outer:
 		logger.With(zap.Error(err)).Errorf("Failed to deep copy external gateway configuration")
 	}
 
+	// Reconcile Slice gateway service type
+	sliceGatewayProvider := workerv1alpha1.WorkerSliceGatewayProvider{
```
Comment (bharath-avesha) [inline service/worker_slice_config_service.go:216]: I was expecting this var: sliceGatewayProvider to be set in the workerSliceConfig obj, but dont see it here. 
Shouldnt this be set to workerSliceConfig.Spec.SliceGatewayProvider?

Comment (mridulgain) [reply service/worker_slice_config_service.go:216]: yes! Rectified.

File: apis/controller/v1alpha1/sliceconfig_types.go [hunk 2]
```diff
@@ -80,10 +80,10 @@ type WorkerSliceGatewayProvider struct {
 
 type SliceGatewayServiceType struct {
 	// +kubebuilder:validation:Required
-	Cluster string `json:"cluster,omitempty"`
```
Comment (bharath-avesha) [inline apis/controller/v1alpha1/sliceconfig_types.go]: I think we should not remove "omitempty". If this config is not specified, we assume that it is NodePort.

Comment (mridulgain) [reply apis/controller/v1alpha1/sliceconfig_types.go]: Currently the behaviour is, if user is defining an array element then both `cluster` & `type` is required else we get `NodePort` for all participating clusters. Please correct me if I understood the comment incorrectly.

**Review by narmidm — APPROVED** [review]

**Pull request merged**